Added Files for new Amazon web scraping script

Nishitbaria opened the pull request:
## Description
This pull request adds a new web scraping script for retrieving product details from Amazon. The script is implemented in Python and utilizes the Beautiful Soup library for HTML parsing. The script includes a class named "Amazon" with several functions for extracting specific product information.

Example Usage:
```
amazon = Amazon("Amazon.html")
amazon.run()
```

The "Amazon" class accepts the path to an HTML file as input. The HTML file should contain the product listings from the Amazon website. The class provides the following functions:

1. `read_html_file()`: Reads the HTML file and parses it using Beautiful Soup.

2. `get_product_name(div)`: Extracts the product name from the given div element.

3. `get_product_price(div)`: Extracts the product price from the given div element.

4. `get_product_reviews(div)`: Extracts the product reviews from the given div element.

5. `get_product_availability(div)`: Extracts the product availability from the given div element.

6. `get_product_image(div)`: Extracts the product image URL from the given div element.

7. `display_product_details()`: Displays the product details by iterating over the divs in the HTML.

8. `run()`: Runs the Amazon class to display the product details.


The `display_product_details()` function iterates over the div elements in the HTML and extracts the desired product information. It then prints the product details if all the required information (name, price, and reviews) is available.

This script provides a convenient way to scrape and retrieve product details from Amazon, which can be further utilized for data analysis, price comparison, or any other purpose.

Please review and merge this pull request if it meets the project requirements.



## Resolves: [#405 ]

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

![image](https://github.com/Clueless-Community/scrape-up/assets/85815172/d890dae2-b034-47ca-9f36-b1719997a994)

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

nikhil25803 reviewed (changes requested):
Again you are pushing html files which is not necessary for the project.
![image](https://github.com/Clueless-Community/scrape-up/assets/93156825/1d3c62bc-2b80-4bc7-9650-b6dd9331a968)

Nishitbaria commented:
@nikhil25803 , i have added html file beacuse web scarping in amazon ,cant be done through url it gives error 503 (The HTTP status code 503 indicates that the server is temporarily unavailable or overloaded and unable to handle the request. It typically occurs when the server is undergoing maintenance or experiencing high traffic, and the client should retry the request later.),so I have implemented approach in which we save webpage in html file ,thats the reason i used html file.

![image](https://github.com/Clueless-Community/scrape-up/assets/85815172/9eab6677-8ad1-478f-8516-95ac57ffcc15)

you can check by running this code : [code](https://replit.com/@jayesheditz/NormalHiddenFile#main.py) ,so to make script working i am using html file.you can check my code it is running

Nishitbaria commented:
@nikhil25803 ,please merge my pull,I have made script from scratch,your earlier script was not working ,I have made it working,We have to use html file as server can handle code to amazon to be parsed everytime,please understand and merge my pull request

nikhil25803 commented:
Your PR is still not valid @Nishitbaria |
These many changes are not acceptable, a lot of scrapping codes are available. Go through them and check how they are implemented in a precise way.

![image](https://github.com/Clueless-Community/scrape-up/assets/93156825/b41e7a0d-1ad3-4fdb-a8cf-6ec61928ab04)

Nishitbaria commented:
@nikhil25803 ,why are you pointing number of lines,if i write whole code from scratch ,more number of lines will be changesd and number of lines are changed because i have added html file which contains  amazon source code,you should understand and please provide me appropriate response and think on merging my request and you can research about amazon scraping ,you will know that a local server cant handle amazon html to be parsed.

nikhil25803 commented:
I am pointing out that again and again because that is the only thing that is not acceptable. 

And about understanding, you should understand that this is a `pip` package, not a repository where you can dump everything you want. Also what you are doing is not an efficient way.  We have covered 20+ platforms from `github` to `Flipkart`. None of them needs a local html file to be parsed.

We get the data from the internet and parse it on the runtime. And if you do not know how to do it, learn it.

Think of an optimized code, make changes, and push the code. Not here to argue, or make you understand again and again. You are not a kid.

Closing this PR for now, and raise only when you have a code that can be integrated with a package.
